cmd/restore.go

```diff
@@ -253,3 +268,35 @@ func (c *restoreCommand) manipulateAgents(operation func(bool) map[string]error)
 	}
 	return nil
 }
+
+const agentConfPattern = "/var/lib/juju/agents/machine-*/agent.conf"
+
+// ReadCredsFromAgentConf tries to load a mongo username and password
+// from the standard agent.conf location on a controller machine.
+func ReadCredsFromAgentConf() (string, string, error) {
+	return ReadCredsFromPattern(agentConfPattern)
+}
+
+// ReadCredsFromPattern tries to load a mongo username and password
+// from the first file it finds matching the pattern passed in.
+func ReadCredsFromPattern(pattern string) (string, string, error) {
+	matches, err := filepath.Glob(pattern)
+	if err != nil {
+		return "", "", errors.Trace(err)
+	}
+	if len(matches) == 0 {
+		return "", "", errors.Errorf("couldn't find an agent.conf - please specify username and password")
+	}
+	conf := matches[0]
```

Comment:
i guess any will do?

Reply:
Yeah, that was my thinking - we'd only expect to find one but it might be 0,1,2...